src/app/shared/add-device/add-device.component.spec.ts

```diff
@@ -0,0 +1,31 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing'
+import { AddDeviceComponent } from './add-device.component'
+import { ProfilesService } from '../../profiles/profiles.service'
+import { of } from 'rxjs'
+
+describe('AddDeviceComponent', () => {
+  let component: AddDeviceComponent
+  let fixture: ComponentFixture<AddDeviceComponent>
+  let getDataSpy: jasmine.Spy
+
+  beforeEach(async () => {
+    const profileService = jasmine.createSpyObj('ProfilesService', ['getData'])
+    getDataSpy = profileService.getData.and.returnValue(of([]))
+    await TestBed.configureTestingModule({
+      declarations: [AddDeviceComponent],
+      providers: [{ provide: ProfilesService, useValue: profileService }]
+    })
+      .compileComponents()
+  })
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(AddDeviceComponent)
+    component = fixture.componentInstance
+    fixture.detectChanges()
+  })
+
+  it('should create', () => {
```

Comment:
Can you add coverage for the rest of the Add-Device component?

Reply:
Added more unit tests.